### PR TITLE
added tests for the reasoner

### PR DESCRIPTION
**Pull request by sklarman**

### CODE REVIEW

**Comment by burukuru** [comment]
retest this please

File: grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java [hunk 1]
```diff
@@ -39,23 +38,185 @@
...
+    @ClassRule
+    public static final GraphContext testSet13 = GraphContext.preLoad("testSet13.gql");
+
     @Before
     public void onStartup() throws Exception {
         assumeTrue(usingTinker());
     }
 
+
+    @Ignore
```
Comment (fppt) [inline grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java]: I think there should be a comment to say why this one is ignored.

File: grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java [hunk 2]
```diff
@@ -39,23 +38,185 @@
...
+    @Test
+    public void generatingMultipleIsaEdges() {
+        QueryBuilder qb = testSet2.graph().graql().infer(true);
+        String query1String = "match $x isa entity2;";
+        QueryAnswers answers1 = queryAnswers(qb.parse(query1String));
+
+        Assert.assertEquals(answers1.size(), 1);
+    }
+
+    @Ignore
```
Comment (fppt) [inline grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java]: Same for here

File: grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java [hunk 3]
```diff
@@ -39,23 +38,185 @@
...
+    public void generatingIsaEdge() {
+        QueryBuilder qb = testSet4.graph().graql().infer(true);
+        String query1String = "match $x isa entity1;";
+        String query2String = "match $x isa entity2;";
+        QueryAnswers answers1 = queryAnswers(qb.parse(query1String));
+        QueryAnswers answers2 = queryAnswers(qb.parse(query2String));
+        Assert.assertTrue(answers1.containsAll(answers2)&&answers2.containsAll(answers1));
+    }
+
+    @Ignore
```
Comment (fppt) [inline grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java]: And here

File: grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java [hunk 4]
```diff
@@ -39,23 +38,185 @@
...
+    @Test
+    public void generatingFreshRelation() {
+        QueryBuilder qb = testSet6.graph().graql().infer(true);
+        String queryString = "match $x isa relation1;";
+        QueryAnswers answers = queryAnswers(qb.parse(queryString));
+
+        Assert.assertEquals(answers.size(), 3);
+    }
+
+    @Ignore
```
Comment (fppt) [inline grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java]: dito

File: grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java [hunk 5]
```diff
@@ -39,23 +38,185 @@
...
+    @Test
+    public void distinctLimitedAnswersOfInfinitelyGeneratingRule() {
+        QueryBuilder qb = testSet7.graph().graql().infer(true);
+        String queryString = "match $x isa relation1; limit 50;";
+        QueryAnswers answers = queryAnswers(qb.parse(queryString));
+        Assert.assertEquals(answers.size(), 50);
+    }
+
+    @Ignore
```
Comment (fppt) [inline grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java]: beep

File: grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java [hunk 6]
```diff
@@ -39,23 +38,185 @@
...
+    @Test
+    public void roleUnificationWithRoleHierarchiesInvolved1() {
+        QueryBuilder qb = testSet8.graph().graql().infer(true);
+        String queryString = "match (role2:$x, role3:$y) isa relation2;";
+        QueryAnswers answers = queryAnswers(qb.parse(queryString));
+        answers.forEach(y -> Assert.assertTrue(y.values().size()<=1));
+    }
+
+    @Ignore
```
Comment (fppt) [inline grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java]: boop

File: grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java [hunk 7]
```diff
@@ -39,23 +38,185 @@
...
+    public void transRelationWithEntityGuardsAtBothEnds() {
         QueryBuilder iqb = testSet10.graph().graql().infer(true);
         String queryString = "match (role1: $x, role2: $y) isa relation2;";
         QueryAnswers answers = queryAnswers(iqb.parse(queryString));
-        Assert.assertThat(answers, not(empty()));
+
+        Assert.assertTrue(!answers.isEmpty());
+    }
+
+    @Ignore
```
Comment (fppt) [inline grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java]: bonk

File: grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java [hunk 8]
```diff
@@ -39,23 +38,185 @@
...
+
+    @Test
+    public void circularRuleDependencies() {
+        QueryBuilder qb = testSet12.graph().graql().infer(true);
+        String queryString = "match (role1:$x, role2:$y) isa relation3;";
+        QueryAnswers answers = queryAnswers(qb.parse(queryString));
+        Assert.assertEquals(answers.size(), 2);
+    }
+
+    @Ignore
```
Comment (fppt) [inline grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java]: bank

**Comment by fppt** [comment]
Good tests names by the way @sklarman. When we break these tests in the future it will make it easier for us to find out why.

**Comment by sklarman** [comment]
@fppt Comments about expected results of each test added.

**Pull request merged**